src/wp-includes/taxonomy.php

```diff
@@ -3751,11 +3743,16 @@ function update_object_term_cache( $object_ids, $object_type ) {
 		}
 	}
 
+	$cache_values = array();
 	foreach ( $object_terms as $id => $value ) {
 		foreach ( $value as $taxonomy => $terms ) {
-			wp_cache_add( $id, $terms, "{$taxonomy}_relationships" );
+			$cache_values[ $taxonomy ][ $id ] = $terms;
+
 		}
 	}
+	foreach ( $cache_values as $taxonomy => $data ) {
```

Comment:
hopefully all this is already tested, it is challenging to follow

Reply:
I saw unit tests failing here - https://github.com/WordPress/wordpress-develop/runs/5067955057?check_suite_focus=true. 

[This](https://github.com/WordPress/wordpress-develop/pull/2271/commits/7181125d72b28b9e9a3e29bb6c8d97c1054dfa5a) change, fixed it.